src/isar/robot/robot.py

```diff
@@ -0,0 +1,261 @@
+            ):
+                self._trigger_event(self.queues.robot_online)
+            self.current_status = robot_status
+            time.sleep(settings.FSM_SLEEP_TIME)
+
+    def run_robot_task_status_thread(self):
+        while True:
+            current_state: Optional[States] = self._check_shared_state(self.queues.state)
+
+            if not current_state in [States.Monitor, States.Paused, States.Stop]:
+                # Here we exit the thread as the mission is done
+                break
+
+            current_task: Optional[Task] = self._check_shared_state(
+                self.queues.state_machine_current_task
+            )
+
+            failed_task_error: Optional[ErrorMessage] = None
+
+            if current_task == None:
+                continue
+
+            task_status: TaskStatus
+            try:
+                task_status = self.robot.task_status(current_task.id)
+                self.request_status_failure_counter = 0
+            except (
+                RobotCommunicationTimeoutException,
+                RobotCommunicationException,
+            ) as e:
+                self.request_status_failure_counter += 1
+
+                if (
+                    self.request_status_failure_counter
+                    >= self.request_status_failure_counter_limit
+                ):
+                    self.logger.error(
+                        f"Failed to get task status "
+                        f"{self.request_status_failure_counter} times because: "
+                        f"{e.error_description}"
+                    )
+                    self.logger.error(
+                        f"Monitoring task failed because: {e.error_description}"
+                    )
+                    failed_task_error = ErrorMessage(
+                        error_reason=e.error_reason,
+                        error_description=e.error_description,
+                    )
+
+                else:
+                    time.sleep(settings.FSM_SLEEP_TIME)
+                    continue
+
+            except RobotTaskStatusException as e:
+                failed_task_error = ErrorMessage(
+                    error_reason=e.error_reason,
+                    error_description=e.error_description,
+                )
+
+            except RobotException as e:
+                failed_task_error = ErrorMessage(
+                    error_reason=e.error_reason,
+                    error_description=e.error_description,
+                )
+
+            if failed_task_error != None:
+                self._trigger_event(
+                    self.queues.robot_task_status_failed,
+                    failed_task_error,
+                )
+                task_status = TaskStatus.Failed
+            else:
+                self._trigger_event(self.queues.robot_task_status, task_status)
+            self.current_status = task_status
+            time.sleep(settings.FSM_SLEEP_TIME)
+
+    def stop(self) -> None:
```

Comment:
when is this triggered?

Reply:
It's not used yet, but I might use it before moving the PR out of draft

Reply:
It is now used in test cleanup